### PR TITLE
feat: add tests

### PR DESCRIPTION
**Pull request by userquin**
This PR includes:
- update pnpm to 8.5.0 and Nuxt 3.4.3
- build and browser tests for `nuxi build and generate`

supersedes #35

### CODE REVIEW

**Comment by userquin** [comment]
@danielroe for Nuxt ecosystem-ci:
```ts
import { runInRepo } from '../utils'
import { RunOptions } from '../types'

export async function test(options: RunOptions) {
	if (options.viteMajor < 4) {
		// only latest version
		return
	}

	await runInRepo({
		...options,
		repo: 'vite-pwa/nuxt',
		branch: 'userquin/feat-add-tests', // <== change to main when merged
		beforeTest: 'pnpm playwright install',
		build: 'prepack',
		test: 'test',
	})
}
```

File: package.json [hunk 1]
```diff
@@ -41,31 +41,40 @@
...
+    "lint-fix": "nr lint --fix",
+    "test:build:serve": "PORT=4173 node playground/.output/server/index.mjs",
+    "test:generate:serve": "PORT=4173 serve playground/dist",
+    "test:build": "nr dev:build && TEST_BUILD=true vitest run && TEST_BUILD=true playwright test",
+    "test:generate": "nr dev:generate && vitest run && playwright test",
+    "test": "nr test:build && nr test:generate",
+    "test:with-build": "nr dev:prepare && nr prepack && nr test"
   },
   "peerDependencies": {
     "@nuxt/kit": "^3.0.0 || ^3.1.0",
```
Comment (danielroe) [inline package.json:58]: should this be bumped?

Comment (userquin) [reply package.json:58]: Can you update the pr?

Comment (danielroe) [reply package.json:58]: I guess my question is whether you specifying 3.0.0 and 3.1.0 separately to work around a package manager issue? It feels like we shouldn't need to specify each minor version with a caret dependency range...

So likely we should either replace with:

```suggestion
    "@nuxt/kit": "^3.0.0",
```

or the much more verbose (if there's a reason for it):

```suggestion
    "@nuxt/kit": "^3.0.0 || ^3.1.0 || ^3.2.0 || ^3.3.0 || ^3.4.0",
```

**Comment by dargmuesli** [comment]
Why superseding my PR :smile: :smiling_face_with_tear:

File: .gitignore [hunk 2]
```diff
@@ -55,3 +55,8 @@ dev-dist/
 Network Trash Folder
 Temporary Items
 .apdisk
+
+# Plawright test
```
Comment (dargmuesli) [inline .gitignore:59]:
```suggestion
# Playwright test
```

**Pull request merged**